modules/dynamic-plugin-api.adoc

```diff
@@ -1475,6 +1471,149 @@ Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Som
+  ]}
+/>
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`customDataSource` |(optional) Base URL of an API endpoint that handles PromQL queries. If provided, this is used instead of the default API for fetching data.
+|`defaultSamples` |(optional) The default number of data samples plotted for each data series. If there are many data series, QueryBrowser might automatically pick a lower number of data samples than specified here.
+|`defaultTimespan` |(optional) The default timespan for the graph in milliseconds - defaults to 1,800,000 (30 minutes).
+|`disabledSeries` |(optional) Disable (don't display) data series with these exact label / value pairs.
+|`disableZoom` |(optional) Flag to disable the graph zoom controls.
+|`filterLabels` |(optional) Optionally filter the returned data series to only those that match these label / value pairs.
+|`fixedEndTime` |(optional) Set the end time for the displayed time range rather than showing data up to the current time.
+|`formatSeriesTitle` |(optional) Function that returns a string to use as the title for a single data series.
+|`GraphLink` |(optional) Component for rendering a link to another page (for example getting more information about this query).
+|`hideControls` |(optional) Flag to hide the graph controls for changing the graph timespan, and so on.
+|`isStack` |(optional) Flag to display a stacked graph instead of a line graph. If showStackedControl is set, it will still be possible for the user to switch to a line graph.
+|`namespace` |(optional) If provided, data is only returned for this namespace (only series that have this namespace label).
+|`onZoom` |(optional) Callback called when the graph is zoomed.
+|`pollInterval` |(optional) If set, determines how often the graph is updated to show the latest data (in milliseconds).
+|`queries` |Array of PromQL queries to run and display the results in the graph.
+|`showLegend` |(optional) Flag to enable displaying a legend below the graph.
+|`showStackedControl` |Flag to enable displaying a graph control for switching between stacked graph mode and line graph mode.
+|`timespan` |(optional) The timespan that should be covered by the graph in milliseconds.
+|`units` |(optional) Units to display on the Y-axis and in the tooltip.
+|===
+
+[discrete]
+== `useAnnotationsModal`
+
+A hook that provides a callback to launch a modal for editing Kubernetes resource annotations.
+
+.Example
+[source,tsx]
+----
+const PodAnnotationsButton = ({ pod }) => {
+  const { t } = useTranslation();
+  const launchAnnotationsModal = useAnnotationsModal<PodKind>(pod);
+  return <button onClick={launchAnnotationsModal}>{t('Edit Pod Annotations')}</button>
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resource` |The resource to edit annotations for an object of K8sResourceCommon type.
+|===
+
+.Returns
+A function which will launch a modal for editing a resource's annotations.
```

Comment:
I think "which" should be "that." Also, we're supposed to avoid "will" (see the [Red Hat Supplemental Style Guide](https://redhat-documentation.github.io/supplementary-style-guide/#_w)) I suggest replacing it with "A function that launches a modal..."

![image](https://github.com/openshift/openshift-docs/assets/83772007/5f0a5754-d1f3-41cc-968c-6dc61e1e9502)

Reply:
These are API's straight from https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/docs/api.md. We would treat these like the REST API docs.